backend/src/nodes/utils/image_utils.py

```diff
@@ -199,31 +199,54 @@ def as_2d_grayscale(img: np.ndarray) -> np.ndarray:
     assert False, f"Invalid image shape {img.shape}"
 
 
-def as_target_channels(img: np.ndarray, target_channels: int) -> np.ndarray:
+def as_3d(img: np.ndarray) -> np.ndarray:
+    """Given a grayscale image, this returns an image with 3 dimensions (image.ndim == 3)."""
+    if img.ndim == 2:
+        return np.expand_dims(img.copy(), axis=2)
+    return img
+
+
+def as_target_channels(
+    img: np.ndarray, target_c: int, narrowing: bool = False
+) -> np.ndarray:
     """
     Given a number of target channels (either 1, 3, or 4), this convert the given image
     to an image with that many channels. If the given image already has the correct
     number of channels, it will be returned as is.
 
-    Only widening conversions are supported.
+    Narrowing conversions are only supported if narrowing is True.
     """
     c = get_h_w_c(img)[2]
 
-    if target_channels == 1:
+    if c == target_c == 1:
         return as_2d_grayscale(img)
-    if c == target_channels:
+    if c == target_c:
         return img
 
-    assert c < target_channels
-
-    if target_channels == 3:
-        if c == 1:
-            return np.dstack((img, img, img))
-
-    if target_channels == 4:
-        return convert_to_BGRA(img, c)
-
-    assert False, "Unable to convert image"
+    if narrowing is False:
```

Comment:
I mean, `is` technically works, but should that really be used over `==` for bools? This is still my first python project, so I'm not really aware of best practices of the language, but seems a bit strange to me.

Reply:
```py
>>> isinstance(True, int)
True
```

You just gave me another good reason to hate python. I never would have expected this. Not even JS is that stupid. Like, I know why they did it, bools in C and all, but come on.